p32/src/bank.rs

```diff
@@ -0,0 +1,249 @@
+                }
+            }
+            match user_option {
+                None => {
+                    return Err(Error::UserNotFound {
+                        username: username_debit,
+                    })
+                }
+                Some(user_d) => match user_d.balance.checked_sub(a) {
+                    Some(b) => {
+                        if b >= 0 || b.unsigned_abs() <= user_d.credit_line {
+                            user_d.balance = b
+                        } else {
+                            return Err(Error::CreditLimitReached);
+                        }
+                    }
+                    None => {
+                        return Err(Error::BalanceUnderflow {
+                            username: username_debit,
+                        })
+                    }
+                },
+            }
+        }
+        {
+            // renewed scope for user_option
+            let mut user_option: Option<&mut User> = None;
+            for user_c in self.users.iter_mut() {
+                if user_c.name == username_credit {
+                    user_option = Some(user_c);
+                    break;
+                }
+            }
+            match user_option {
+                None => {
+                    result = Err(Error::UserNotFound {
+                        username: username_credit,
+                    });
+                    reset_flag = true
+                }
+                Some(user_c) => match user_c.balance.checked_add(a) {
+                    Some(b) => user_c.balance = b,
+                    None => {
+                        result = Err(Error::BalanceOverflow {
+                            username: username_credit,
+                        });
+                        reset_flag = true
+                    }
+                },
+            }
+        }
+        if reset_flag {
+            for user_d in self.users.iter_mut() {
+                if user_d.name == username_debit {
+                    user_d.balance += a;
+                    return result;
+                }
+            }
+        }
+        result
+    }
+
+    // Transfer funds rewritten using indices
+    pub fn transfer_funds_alternate(
+        &mut self,
+        username_debit: String,
+        username_credit: String,
+        amount: u64,
+    ) -> Result<(), Error> {
+        let a = match i64::try_from(amount) {
+            Ok(am) => am,
+            Err(..) => return Err(Error::BadAmount { amount }),
+        };
+        let mut index_d_option: Option<usize> = None;
+        let mut index_c_option: Option<usize> = None;
+        for (index, user) in self.users.iter().enumerate() {
+            if user.name == username_debit {
+                index_d_option = Some(index);
+            }
+            if user.name == username_credit {
+                index_c_option = Some(index);
+            }
+            if let (Some(..), Some(..)) = (index_d_option, index_c_option) {
```

Comment:
In cases like this usually we use `_` instead of `..`, i.e.: `if let (Some(_), Some(_)) = (index_d_option, index_c_option) { .. }`. Both have the same effect, but underscore is much more common.